providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/grpc/GrpcResolver.java

```diff
@@ -160,28 +165,33 @@ private <ValT, ReqT extends Message, ResT extends Message> ProviderEvaluation<Va
                 .build();
 
         // run the referenced resolver method
-        Message response = strategy.resolve(resolverRef, req, key);
-
-        // parse the response
-        ValT value = converter == null ? getField(response, VALUE_FIELD)
-                : converter.convert(getField(response, VALUE_FIELD));
-
-        // Extract metadata from response
-        ImmutableMetadata immutableMetadata = metadataFromResponse(response);
-
-        ProviderEvaluation<ValT> result = ProviderEvaluation.<ValT>builder()
-                .value(value)
-                .variant(getField(response, VARIANT_FIELD))
-                .reason(getField(response, REASON_FIELD))
-                .flagMetadata(immutableMetadata)
-                .build();
+        try {
+            Message response = strategy.resolve(resolverRef, req, key);
+
+            // parse the response
+            ValT value = converter == null ? getField(response, VALUE_FIELD)
+                    : converter.convert(getField(response, VALUE_FIELD));
+
+            // Extract metadata from response
+            ImmutableMetadata immutableMetadata = metadataFromResponse(response);
+
+            ProviderEvaluation<ValT> result = ProviderEvaluation.<ValT>builder()
+                    .value(value)
+                    .variant(getField(response, VARIANT_FIELD))
+                    .reason(getField(response, REASON_FIELD))
+                    .flagMetadata(immutableMetadata)
+                    .build();
+
+            // cache if cache enabled
+            if (this.isEvaluationCacheable(result)) {
+                this.cache.put(key, result);
+            }
 
-        // cache if cache enabled
-        if (this.isEvaluationCacheable(result)) {
-            this.cache.put(key, result);
+            return result;
+        } catch (Exception e) {
```

Comment:
minor refactoring - shall we simplify catch to where the exception is thrown 

```Java

        try {
            Message response = strategy.resolve(resolverRef, req, key);
         }catch(){
          //   ... handling 
         }   
         
        // rest of the logic
```            

in my view this helps to understand the logic better

Reply:
We could, but then we'd have to declare the `response` var outside the try block, which I tend not to like because it introduces more mutability.

On the other hand, it might be more readable. Do you think it's better to have a null/mutated `response` var and keep the try/catch smaller?

Reply:
We can make `Message response` effectively final, 

```java
        final Message response;
        try {
            response = strategy.resolve(resolverRef, req, key);
        } catch (Exception e) {
            OpenFeatureError openFeatureError = mapError(e);
            throw openFeatureError;
        }
```

personally I think it makes code cleaner and more readable :) anyway let's not worry much about this.

Reply:
I will make this change.

Reply:
Done!